client/ayon_core/pipeline/load/plugins.py

```diff
@@ -264,6 +266,43 @@ class ProductLoaderPlugin(LoaderPlugin):
     """
 
 
+class PreLoaderHookPlugin:
+    """Plugin that should be run before any Loaders in 'loaders'
+
+    Should be used as non-invasive method to enrich core loading process.
+    Any external studio might want to modify loaded data before or afte
+    they are loaded without need to override existing core plugins.
+    """
+    loaders: ClassVar[set[str]]
+
+    def process(self, context, name=None, namespace=None, options=None):
+        pass
+
+    def update(self, container, context):
+        pass
+
+    def switch(self, container, context):
+        pass
+
+
+class PostLoaderHookPlugin:
```

Comment:
Should Post Loader hook plug-in get access to the loaded container? Or how would one actually detect _what_ was loaded?

Reply:
Added container argument